File: haushaltsbuch-api/src/main/java/com/github/haschi/haushaltsbuch/api/BeginneHaushaltsbuchführung.java
package com.github.haschi.haushaltsbuch.api;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.github.haschi.modeling.de.Anweisung;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

import java.util.UUID;

@Anweisung("Beginne Haus&shy;halts&shy;buch&shy;füh&shy;rung")
@JsonSerialize(as = ImmutableBeginneHaushaltsbuchführung.class)
@JsonDeserialize(as = ImmutableBeginneHaushaltsbuchführung.class)
public interface BeginneHaushaltsbuchführung
{
    @TargetAggregateIdentifier
    UUID id();
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/modell/core/commands/BeginneHaushaltsbuchführung.kt
package org.github.haschi.haushaltsbuch.modell.core.commands

import org.axonframework.commandhandling.TargetAggregateIdentifier
import org.github.haschi.haushaltsbuch.modell.core.values.Inventar
import org.github.haschi.haushaltsbuch.modell.core.values.Aggregatkennung

data class BeginneHaushaltsbuchführung(
        @TargetAggregateIdentifier val id: Aggregatkennung,
        val inventar: Inventar)
